### PR TITLE
[Modules] Updated Insight DiagnosticSettings to new dependencies approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated Insight DiagnosticSettings to new dependencies approach
- Added Set-ModuleReadMe capabiltity to deal with empty bicep parameters
- Enhanced robustness of validation script

| Pipeline |
| - |
| [Insights: DiagnosticSettings](https://github.com/Azure/ResourceModules/actions/workflows/ms.insights.diagnosticsettings.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Insights/diagnosticSettings/.test/common/dependencies.bicep [hunk 1]
```diff
@@ -0,0 +1,3 @@
+@description('Optional. The location to deploy to.')
```
Comment (eriqua) [inline modules/Microsoft.Insights/diagnosticSettings/.test/common/dependencies.bicep:1]: Please remove if not needed

File: modules/Microsoft.Insights/diagnosticSettings/.test/common/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+// =========== //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+module resourceGroupResources 'dependencies.bicep' = {
```
Comment (eriqua) [inline modules/Microsoft.Insights/diagnosticSettings/.test/common/deploy.test.bicep:27]: Please remove if not needed

File: utilities/tools/Set-ModuleReadMe.ps1 [hunk 3]
```diff
@@ -650,6 +649,11 @@ function ConvertTo-FormattedJSONParameterObject {
         [string] $BicepParamBlock
     )
 
+    if ([String]::IsNullOrEmpty($BicepParamBlock)) {
```
Comment (eriqua) [inline utilities/tools/Set-ModuleReadMe.ps1:652]: Is this change related to this module testing update?

Comment (MrMCake) [reply utilities/tools/Set-ModuleReadMe.ps1:652]: It is. A fix was merged to main and I opened a follow-up PR #2133 to main too - that should resolve this conflict

File: modules/Microsoft.Insights/diagnosticSettings/.test/common/dependencies.bicep (hunk 1 above)
Comment (MrMCake) [reply modules/Microsoft.Insights/diagnosticSettings/.test/common/dependencies.bicep:1]: done

File: modules/Microsoft.Insights/diagnosticSettings/.test/common/deploy.test.bicep (hunk 2 above)
Comment (MrMCake) [reply modules/Microsoft.Insights/diagnosticSettings/.test/common/deploy.test.bicep]: done

File: utilities/tools/Set-ModuleReadMe.ps1 (hunk 3 above)
Comment (eriqua) [reply utilities/tools/Set-ModuleReadMe.ps1:652]: Unfortunately it didn't. Bad boys merge conflicts to be solved.

Comment (MrMCake) [reply utilities/tools/Set-ModuleReadMe.ps1:652]: resolved

Comment (eriqua) [reply utilities/tools/Set-ModuleReadMe.ps1:652]: Sorry to insist. I still can't see a reason why an update to the set-modulereadme script should be part of a PR covering the new dependencies approach of a module :/

Comment (MrMCake) [reply utilities/tools/Set-ModuleReadMe.ps1:652]: Ah Pardon, I mixed this up with the Pester Test File Update

Comment (MrMCake) [reply utilities/tools/Set-ModuleReadMe.ps1:652]: Opened PR #2134 
> Note: You'll likely see this change in other PRs too (until I merge the PR back from main) as this bug effectively blocked the pipelines

**Review by eriqua — CHANGES_REQUESTED** [review]
It looks like the min deployment test causes the deployment to hang. I'd suggest to remove the min test it from this PR and open a separate issue to investigate that. Mostly because it could be that required parameters, and so the module, need to be updated.

File: modules/Microsoft.Insights/diagnosticSettings/.test/common/deploy.test.bicep [hunk 4]
```diff
@@ -0,0 +1,55 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes')
```
Comment (eriqua) [inline modules/Microsoft.Insights/diagnosticSettings/.test/common/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.Insights/diagnosticSettings/.test/common/deploy.test.bicep [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+@maxLength(90)
+param resourceGroupName string = 'ms.insights.diagnosticsettings-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to')
```
Comment (eriqua) [inline modules/Microsoft.Insights/diagnosticSettings/.test/common/deploy.test.bicep:10]:
```suggestion
@description('Optional. The location to deploy resources to.')
```

Comment (eriqua) [inline modules/Microsoft.Insights/diagnosticSettings/.test/common/deploy.test.bicep:10]:
```suggestion
@description('Optional. The location to deploy resources to.')
```

File: modules/Microsoft.Insights/diagnosticSettings/.test/common/deploy.test.bicep [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints')
```
Comment (eriqua) [inline modules/Microsoft.Insights/diagnosticSettings/.test/common/deploy.test.bicep:13]:
```suggestion
@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
```

**Review by eriqua — APPROVED** [review]
🆗
